Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncPerformanceTestContext.swift

```diff
@@ -0,0 +1,247 @@
+            (systemAttributes[FileAttributeKey.systemFreeSize] as? NSNumber)?.doubleValue ?? 0.0
+        let diskUsed = freeSpaceBefore - freeSpaceAfter
+
+        // Perform custom taredown if specified
+        try teardown(self, numDocs, docSize)
+
+        return IterationResult(time: timeMs,
+                               networkSentBytes: networkSent,
+                               networkReceivedBytes: networkReceived,
+                               cpu: pointInTimeMetrics.cpuData,
+                               memory: pointInTimeMetrics.memoryData,
+                               disk: diskUsed,
+                               threads: pointInTimeMetrics.threadData)
+    }
+
+    func waitForAllStreamsOpen() {
+        while !coll.sync.proxy.dataSynchronizer.allStreamsAreOpen {
+            print("waiting for all streams to open before doing sync pass")
+            sleep(1)
+        }
+    }
+
+    func streamAndSync() throws {
+        if  harness.networkMonitor.state == .connected {
+            // add the stream joiner as a delegate of the stream so we can wait for events
+            if let iCSDel = coll.sync.proxy
+                .dataSynchronizer
+                .instanceChangeStreamDelegate,
+                let nsConfig = iCSDel[MongoNamespace(databaseName: dbName, collectionName: collName)] {
+                nsConfig.add(streamDelegate: streamJoiner)
+                streamJoiner.streamState = nsConfig.state
+            }
+
+            // wait for streams to open
+            waitForAllStreamsOpen()
+        }
+        _ = try coll.sync.proxy.dataSynchronizer.doSyncPass()
+    }
+
+    func powerCycleDevice() throws {
+        try coll.sync.proxy.dataSynchronizer.reloadConfig()
+        if streamJoiner.streamState != nil {
+            streamJoiner.wait(forState: .closed)
+        }
+    }
+
+    func clearLocalDB() throws {
+        // swiftlint:disable force_cast
+        coll.sync.desync(ids: coll.sync.syncedIds().map({$0.value}))
+        try CoreLocalMongoDBService.shared.localInstances.forEach { client in
+            do {
+                try client.listDatabases().forEach {
+                    try? client.db($0["name"] as! String).drop()
+                }
+            } catch {
+                throw "Could not drop databases in ProductionPerformanceTestContext.teardown()"
+            }
+        }
+        // swiftlint:enable force_cast
+    }
+}
+
+func getMemoryUsage() -> Double {
+    var taskInfo = mach_task_basic_info()
+    var count = mach_msg_type_number_t(MemoryLayout<mach_task_basic_info>.size)/4
+    let kerr: kern_return_t = withUnsafeMutablePointer(to: &taskInfo) {
+        $0.withMemoryRebound(to: integer_t.self, capacity: 1) {
+            task_info(mach_task_self_, task_flavor_t(MACH_TASK_BASIC_INFO), $0, &count)
+        }
+    }
+
+    if kerr == KERN_SUCCESS {
+        return Double(taskInfo.resident_size)
+    } else {
+        print("Error getting memory")
+        return 0.0
```

Comment:
[q][minor] Maybe this should be a fatal error? This would completely skew results if some tests have memory usage and others do not.